cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+                        .length(storageSegment.getLength())
+                        .sealed(storageSegment.isSealed())
+                        .deleted(storageSegment.isActive())
+                        .lastModified(new ImmutableDate(storageSegment.getLastModified()))
+                        .attributes(attribs)
+                        .build();
+
+                MetadataStore.SegmentInfo sereializedContainerSegment = MetadataStore.SegmentInfo.builder()
+                        .segmentId(segmentInfo.getSegmentId())
+                        .properties(segmentProperties)
+                        .build();
+                TableEntry unversionedEntry = TableEntry.unversioned(segmentEntry.get(0).getKey().getKey(), SERIALIZER.serialize(sereializedContainerSegment));
+                extension.put(NameUtils.getMetadataSegmentName(container.getId()), Collections.singletonList(unversionedEntry), TIMEOUT).join();
+            }
+        }
+    }
+
+    /**
+     * Segregates container and storage metadata chunks.
+     */
+    private Map<String, List<File>> segregateMetadataSegments(File[] chunkFiles) {
+        Map<String, List<File>> metadataSegments = new HashMap<>();
+        Arrays.stream(chunkFiles)
+                .filter(File::isFile)
+                .filter(f -> !f.getName().contains(ATTRIBUTE_SUFFIX))
+                .forEach(file -> {
+                    String segment = file.getName().split(EPOCH_SPLITTER)[0];
+                    metadataSegments.putIfAbsent(segment, new ArrayList<File>());
+                    metadataSegments.get(segment).add(file);
+                });
+        return metadataSegments;
+    }
+
+    /**
+     * Writes all the parsed TableSegment operations from raw chunks into container_metadata of the debug
+     * segment container.
+     * @param container Container whose container_metadata table segment is populated.
+     * @param tableSegment name of container metadata segment.
+     * @param tableSegmentOperations Operations parsed from raw chunk bytes for container metadata.
+     * @throws Exception
+     */
+    private void writeEntriesToContainerMetadata(DebugStreamSegmentContainer container, String tableSegment, List<TableSegmentUtils.TableSegmentOperation> tableSegmentOperations) throws Exception {
+        output("Writing entries to container metadata");
+        ContainerTableExtension tableExtension = container.getExtension(ContainerTableExtension.class);
+        HashSet<String> deletedKeys = new HashSet<>();
+        for (TableSegmentUtils.TableSegmentOperation operation : tableSegmentOperations) {
+            TableSegmentEntry entry = operation.getContents();
+            TableEntry unversionedEntry = TableEntry.unversioned(new ByteBufWrapper(entry.getKey().getKey()), new ByteBufWrapper(entry.getValue()));
+            String seg = new String(unversionedEntry.getKey().getKey().getCopy());
+
+            if (!allowSegment(seg)) {
```

Comment:
comment will help understand what this check in doing.

Reply:
added comment.